File: client-clojure/src/sirqul_io_t_platform/specs/queue_response.clj
(ns sirqul-io-t-platform.specs.queue-response
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            )
  (:import (java.io File)))


(def queue-response-data
  {
   (ds/opt :queueId) int?
   (ds/opt :name) string?
   (ds/opt :host) string?
   (ds/opt :port) int?
   (ds/opt :username) string?
   (ds/opt :password) string?
   (ds/opt :virtualHost) string?
   (ds/opt :workers) int?
   (ds/opt :exchanger) string?
   (ds/opt :exchangerType) string?
   (ds/opt :handleDelivery) boolean?
   (ds/opt :dataMapping) string?
   (ds/opt :analyticTags) string?
   })

(def queue-response-spec
  (ds/spec
    {:name ::queue-response
     :spec queue-response-data}))

File: client-scalaz/src/main/scala/org/openapitools/client/api/QueueResponse.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import QueueResponse._

case class QueueResponse (
  queueId: Option[Long],
name: Option[String],
host: Option[String],
port: Option[Integer],
username: Option[String],
password: Option[String],
virtualHost: Option[String],
workers: Option[Integer],
exchanger: Option[String],
exchangerType: Option[String],
handleDelivery: Option[Boolean],
dataMapping: Option[String],
analyticTags: Option[String])

object QueueResponse {
  import DateTimeCodecs._

  implicit val QueueResponseCodecJson: CodecJson[QueueResponse] = CodecJson.derive[QueueResponse]
  implicit val QueueResponseDecoder: EntityDecoder[QueueResponse] = jsonOf[QueueResponse]
  implicit val QueueResponseEncoder: EntityEncoder[QueueResponse] = jsonEncoderOf[QueueResponse]
}
